src/LinearSolve.jl

```diff
@@ -0,0 +1,59 @@
+using LinearAlgebra
+
+export safe_linear_solve, safe_linear_solve!, add_diagonal_regularization!
+
+"""
+    safe_linear_solve(A, b; warn_on_singular=true)
+
+Solves the linear system Ax = b with robust handling of ill-conditioned matrices.
+
+# Arguments
+- `A`: Coefficient matrix
+- `b`: Right-hand side vector or matrix  
+- `warn_on_singular`: Whether to issue warnings when ill-conditioned matrices are detected
+
+"""
+function safe_linear_solve(A::AbstractMatrix, b::AbstractVecOrMat; warn_on_singular = true)
```

Comment:
Although this can be used, it appears that `warn_on_singular` is not made settable.
I'd say maybe for now we can link it to "get_verbose(ekp)"?